### PR TITLE
Fix incorrect gyro bias estimation algorithm - sin/cos swapped

### PR DESCRIPTION
**Pull request by evanmayer**
See comment for explanation.

### CODE REVIEW

**Comment by evanmayer** [comment]
Oops, lock pin changed got pushed onto this.

**Comment by evanmayer** [comment]
### Lock pin changes

* Update preamble to listen to attached limit switches
* Update motion commands to perform an infinite pos/neg move to extend/retract the lock pin, guarded by the limit switches
* Update unit tests
* Keep old functionality until we have a chance to test with hardware in Ft. Sumner

### Lock Pin Todos:

* Need to update acceptable lock pin positions based on our lock pin plate, and actual encoder readings once test flight gondola is assembled
* Need to test polarity of move commands and mapping of `?4` query bit field to limit switches

File: mcp/motors/actuators.c [hunk 1]
```diff
@@ -993,10 +1003,289 @@ static void DoShutter(void)
...
+        retval = -1;
+    }
+    return retval;
+}
+
+/**
+ * @brief Update structs related to elevation lock.
+ * The NiC MCC does this via the BlastBus to give it a chance to know what's
+ * going on. The ICC reads it directly to get more promptly the answer (since
+ * all these fields are slow).
```
Comment (ianlowe13) [inline mcp/motors/actuators.c:1037]: i feel like this comment should be updated, BLASTBus was a physical data bus card network.

Comment (ianlowe13) [reply mcp/motors/actuators.c:1037]: we now use a data sharing server

File: mcp/motors/actuators.c [hunk 2]
```diff
@@ -993,10 +1003,289 @@ static void DoShutter(void)
...
+ * @param nic flag to determine where the state data is queried from. 1 =
+ * other computer, 0 = in-charge computer
```
Comment (ianlowe13) [inline mcp/motors/actuators.c:1039]: wow i hate that this needs to take something like ``!InCharge`` as an argument rather than ``InCharge``

**Review by ianlowe13 — CHANGES_REQUESTED** [review]

File: mcp/motors/actuators.c (hunk 1 above)
Comment (evanmayer) [reply mcp/motors/actuators.c:1037]: Updated

File: blast_config/tx_struct_tng.c
[the remark this answers is not in the document]
Comment (evanmayer) [reply blast_config/tx_struct_tng.c:1]: Fixed

File: mcp/commanding/commands.c
[the remark this answers is not in the document]
Comment (evanmayer) [reply mcp/commanding/commands.c:1]: Fixed

**Comment by evanmayer** [comment]
Closes #103

**Review by ianlowe13 — APPROVED** [review]

**Pull request merged**